velox/exec/FilterProject.cpp

```diff
@@ -184,22 +164,28 @@ RowVectorPtr FilterProject::getOutput() {
     if (!allRowsSelected) {
       rows->setFromBits(filterEvalCtx_.selectedBits->as<uint64_t>(), size);
     }
-    project(*rows, evalCtx);
+    project(*rows, evalCtx, results);
   }
 
   return fillOutput(
-      numOut, allRowsSelected ? nullptr : filterEvalCtx_.selectedIndices);
+      numOut,
+      allRowsSelected ? nullptr : filterEvalCtx_.selectedIndices,
+      results);
 }
 
-void FilterProject::project(const SelectivityVector& rows, EvalCtx& evalCtx) {
+void FilterProject::project(
+    const SelectivityVector& rows,
+    EvalCtx& evalCtx,
+    std::vector<VectorPtr>& results) {
   exprs_->eval(
-      hasFilter_ ? 1 : 0, numExprs_, !hasFilter_, rows, evalCtx, results_);
+      hasFilter_ ? 1 : 0, numExprs_, !hasFilter_, rows, evalCtx, results);
 }
 
 vector_size_t FilterProject::filter(
     EvalCtx& evalCtx,
-    const SelectivityVector& allRows) {
-  exprs_->eval(0, 1, true, allRows, evalCtx, results_);
-  return processFilterResults(results_[0], allRows, filterEvalCtx_, pool());
+    const SelectivityVector& allRows,
+    std::vector<VectorPtr>& results) {
```

Comment:
Looks like there is no need to pass 'results' to this method. It can simply use a local variable. This would simplify the API.

Reply:
Hi @mbasmanova, thank you for reviewing! The reason I'm passing a `std::vector<VectorPtr>& results` is that the caller of filter() and project() needs to have access to their results and pass results to fillOutput().

Reply:
Discussed offline. filter() doesn't need to return the results through a std::vector<VectorPtr>. project() can create a local std::vector<VectorPtr> and return it. Updated PR accordingly.